docs/docs/tutorials/jwt_authentication.md

```diff
@@ -0,0 +1,515 @@
+
+```dart
+import 'dart:io';
+
+import 'package:authenticated_app/user.dart';
+import 'package:dart_frog/dart_frog.dart';
+
+Future<Response> onRequest(RequestContext context) {
+  return switch (context.request.method) {
+    HttpMethod.post => _onPost(context),
+    _ => Future.value(Response(statusCode: HttpStatus.methodNotAllowed)),
+  };
+}
+
+Future<Response> _onPost(RequestContext context) async {
+  final task = await context.request.body();
+  final user = context.read<User>();
+
+  return Response.json(
+    body: {
+      'recorded_task': task,
+      'user_id': user.id,
+    },
+  );
+}
+```
+
+## Protecting the authentication token
+
+This is a good point to review what we have done so far:
+
+- We have created a sign in route, where credentials can be posted, and an authentication token
+  is returned if valid.
+- We have routes that can only be accessed if an authentication token is sent in the request.
+
+But like we noticed in the steps before, our authentication token is quite unsafe, it is nothing
+more than the user's username, meaning that if an ill intentioned person guesses another user's
+username, which is not something hard to do, they could pass as that user and maybe steal
+information or other bad actions.
+
+To avoid that, we have to make our tokens in a way where they cannot be faked, guessed or tampered.
+There are many ways of doing that, in this tutorial, we will use JWT, a widely used standard in the
+industry to secure issued tokens. This tutorial will not go much in deep on how JWT tokens work
+under the hood, so to get a better understanding on how they work, be sure to check their official
+[documentation](https://jwt.io/).
+
+Right, with that brief introduction, lets get that done. Luckily, the Dart ecosystem already have
+a handy package that makes it easy to work with JWT tokens, lets start by adding that dependency to our
+project:
+
+```dart
+dart pub add dart_jsonwebtoken
+```
+
+Next, lets add the following method to our `Authenticator` class:
+
+```dart
+  String generateToken({
+    required String username,
+    required User user,
+  }) {
+    final jwt = JWT(
+      {
+        'id': user.id,
+        'name': user.name,
+        'username': username,
+      },
+    );
+
+    return jwt.sign(SecretKey('123'));
+  }
+```
+
+That will take care of generating a JWT token. Note that we call a method `sign` out of the token,
+passing a secret key, as the name implies, this key is secret and should be kept as such, in this
+tutorial we are keeping it hard coded for the sake of simplicity, but in a real case application,
+be sure to correctly store it and pass it to the code in a way where then will remain secret to outsiders.
+
+The sign method will create a signature out of the data we passed to it, that signature will be part
+of the token and will later on allow us to check if an authentication token that we've received
+is valid and if hasn't been tampered!
```

Comment:
```suggestion
The `sign` method will create a signature out of the data we passed to it, ant it will be part
of the token. This technique will allow us to check if an authentication token that we've received
is valid and if it hasn't been tampered!
```